web/src/app/util/PageActions.js

```diff
@@ -8,104 +8,96 @@ const PageActionsContext = React.createContext({
 })
 PageActionsContext.displayName = 'PageActionsContext'
 
-export class PageActionContainer extends React.PureComponent {
-  render() {
-    return (
-      <PageActionsContext.Consumer>
-        {({ actions }) => actions}
-      </PageActionsContext.Consumer>
-    )
-  }
+export function PageActionContainer() {
+  return (
+    <PageActionsContext.Consumer>
+      {({ actions }) => actions}
+    </PageActionsContext.Consumer>
+  )
 }
 
-export class PageActionProvider extends React.PureComponent {
-  state = {
-    actions: null,
-  }
+export function PageActionProvider(props) {
+  // state = {
+  //   actions: null,
+  // }
+  const [actions, setActions] = useState(null)
 
-  _mountCount = 0
-  _mounted = false
+  let _mountCount = 0
+  let _mounted = false
+  let _pending = null
 
-  componentDidMount() {
-    this._mounted = true
-    if (this._pending) {
-      this.setActions(this._pending)
-      this._pending = null
+  function _handleSetActions(actns) {
+    if (!_mounted) {
+      _pending = actns
+      return
     }
-  }
 
-  componentWillUnmount() {
-    this._mounted = false
-    this._pending = false
-    this.setActions.cancel()
+    setActions(actns)
   }
 
-  _setActions = (actions) => {
-    if (!this._mounted) {
-      this._pending = actions
-      return
-    }
+  const handleSetActions = debounce(_handleSetActions)
 
-    this.setState({ actions })
-  }
+  useEffect(() => {
+    _mounted = true
+    if (_pending) {
+      handleSetActions(_pending)
+      _pending = null
+    }
 
-  setActions = debounce(this._setActions)
+    return () => {
+      _mounted = false
+      _pending = false
+      handleSetActions.cancel()
+    }
+  }, [])
 
-  updateMounted = (mount) => {
+  function updateMounted(mount) {
     if (mount) {
-      this._mountCount++
+      _mountCount++
     } else {
-      this._mountCount--
+      _mountCount--
     }
 
-    if (this._mountCount > 1 && global.console && console.error) {
+    if (_mountCount > 1 && global.console && console.error) {
       console.error(
         'PageActions: Found more than one <PageActions> component mounted within the same provider.',
       )
     }
   }
 
-  render() {
-    return (
-      <PageActionsContext.Provider
-        value={{
-          actions: this.state.actions,
-          setActions: this.setActions,
-          trackMount: this.updateMounted,
-        }}
-      >
-        {this.props.children}
-      </PageActionsContext.Provider>
-    )
-  }
+  return (
+    <PageActionsContext.Provider
+      value={{
+        actions: actions,
+        setActions: handleSetActions,
+        trackMount: updateMounted,
+      }}
+    >
+      {props.children}
+    </PageActionsContext.Provider>
+  )
 }
 
-class PageActionUpdater extends React.PureComponent {
-  static propTypes = {
-    setActions: p.func.isRequired,
-  }
-
-  _mounted = false
+function PageActionUpdater(props) {
+  let _mounted = false
```

Comment:
Same as above, each render will potentially have this reset -- we'd only have it shared during the useEffect and its subsequent cleanup call, but future renders will lose the value.